astropy/time/core.py

```diff
@@ -374,6 +377,17 @@ def __new__(cls, val, val2=None, format=None, scale=None,
                 precision=None, in_subfmt=None, out_subfmt=None,
                 location=None, copy=False):
 
+        # Because of import problems, this can only be done on
+        # first call of Time.
+        global _LEAP_SECONDS_CHECKED
+        if not _LEAP_SECONDS_CHECKED:
```

Comment:
Maybe you could save two lines of code and a `global` by holding this attribute in the `Time` class.  So
```
if not getattr(Time, '_LEAP_SECONDS_CHECKED', False):
    Time._LEAP_SECONDS_CHECKED = True
    update_leap_seconds()
```
But on the whole this solution seems fine.  I did confirm that it resolves the problems I found.

Reply:
I thought about putting it on the class, but ended up feeling it was slightly more elegant (less inelegant?) to have it as global state,. Also perhaps slightly faster?